src/modules/Blur/Blur.js

```diff
@@ -1,84 +1,136 @@
 module.exports = exports = function(pixels, blur) {
-    let kernel = kernelGenerator(blur, 1), oldpix = require('lodash').cloneDeep(pixels);
-    kernel = flipKernel(kernel);
-
-    for (let i = 0; i < pixels.shape[0]; i++) {
-        for (let j = 0; j < pixels.shape[1]; j++) {
-            let neighboutPos = getNeighbouringPixelPositions([i, j]);
-            let acc = [0.0, 0.0, 0.0, 0.0];
-            for (let a = 0; a < kernel.length; a++) {
-                for (let b = 0; b < kernel.length; b++) {
-                    acc[0] += (oldpix.get(neighboutPos[a][b][0], neighboutPos[a][b][1], 0) * kernel[a][b]);
-                    acc[1] += (oldpix.get(neighboutPos[a][b][0], neighboutPos[a][b][1], 1) * kernel[a][b]);
-                    acc[2] += (oldpix.get(neighboutPos[a][b][0], neighboutPos[a][b][1], 2) * kernel[a][b]);
-                    acc[3] += (oldpix.get(neighboutPos[a][b][0], neighboutPos[a][b][1], 3) * kernel[a][b]);
-                }
-            }
-            pixels.set(i, j, 0, acc[0]);
-            pixels.set(i, j, 1, acc[1]);
-            pixels.set(i, j, 2, acc[2]);
-        }
+  let kernel = kernelGenerator(blur);
```

Comment:
This is awesome. Should we think about keeping the original as a fallback for environments where the new version wouldn't work? Are there any? Like a device with no GPU?

Thanks!

Reply:
@jywarren GPU.js has an internal fallback but it is not working which is the bug that Robert is fixing. It will mostly get fixed in the next rc(which will release today).